py-polars/polars/functions/as_datatype.py

```diff
@@ -283,14 +283,14 @@ def time_(
 
 def duration(
     *,
-    weeks: Expr | str | int | None = None,
-    days: Expr | str | int | None = None,
-    hours: Expr | str | int | None = None,
-    minutes: Expr | str | int | None = None,
-    seconds: Expr | str | int | None = None,
-    milliseconds: Expr | str | int | None = None,
-    microseconds: Expr | str | int | None = None,
-    nanoseconds: Expr | str | int | None = None,
+    weeks: Expr | str | int | float | None = None,
```

Comment:
As you rightly identified, we shouldn't accept float values for duration components - so also shouldn't imply that we do by adding it in to the type signature. If you want it so the tests doesn't fail lint, can just use a lint directive in the test instead of extending it here 👍

Reply:
I don't understand you @alexander-beedie . The goal of this PR is to _do_ accept `float` values, why shouldn't the signature reflect that?

Reply:
In the linked Issue we hadn't actually agreed on the way forward; @MarcoGorelli (and myself - I just commented) are both of the opinion that we should *not*  allow fractional duration components, as some floating point values are inevitably going to give slightly "off" results, where exact/correct values would be possibly simply by specifying the smaller integer units (minutes ,seconds, etc).

I thought you were taking that approach here ;)